### ISSUE TITLE
Large difference in read time pylas vs laspy

### ISSUE THREAD

**Issue by chrise96** [laspy/laspy#152]
I want to migrate a project from pylas to laspy. However, it seems like the reading time of a laz file takes in this example 10 times longer with laspy.

The code:
```Python
import time
import pylas
import laspy

start = time.time()
in_file = 'C_25DN2.LAZ'
las = laspy.read(in_file)
end = time.time()
print(f'laspy read in {end-start:.2f} seconds.')

start = time.time()
in_file = 'C_25DN2.LAZ'
las = pylas.read(in_file)
end = time.time()
print(f'pylas read in {end-start:.2f} seconds.')
```
Output:
```
laspy read in 570.93 seconds.
pylas read in 53.52 seconds.
```

The laz file that I use can be downloaded here (around 3GB): [https://download.pdok.nl/rws/ahn3/v1_0/laz/C_25DN2.LAZ](https://download.pdok.nl/rws/ahn3/v1_0/laz/C_25DN2.LAZ)

Versions:
lazrs==0.1.1 and pylas==0.4.3
lazrs==0.3.0 and laspy==2.0.1

**Comment by tmontaigu** [comment]
These numbers are strange

I can't do laspy.read or pylas.read on the file as it is too big

However If I read the file by chunk:
```python
in_file = r"C_25DN2.LAZ"
start = time.time()
count = 0
with laspy.open(in_file) as las:
    point_count = las.header.point_count
    for p in las.chunk_iterator(40_000_000):
        count += len(p)
end = time.time()
print(f'laspy read in {end-start:.2f} seconds.')
assert count == point_count
```
It prints `laspy read in 78.19 seconds.` which is far from the 570 seconds you seem to have.

lazrs 0.2.0 (the python bindings) introduced the ability to read/write directly using python file objects instead of having to read all the LAZ file content in memory before being able to decompress it. 
This allows laspy to read / write chunk by chunk to be able to write scripts that handle big files using the new laz-rs 0.4.0 (the rust lib) `ParLasZipDecompressor`.

However the implementation of `ParLasZipDecompressor` seems to use a lot of memory internally to be able to work, so the bigger the file you want to decompress is, the higher its memory usage is.

# TLDR
Which brings to what I think is the cause of the problem lazrs's new `ParLasZipDecompressor` uses a lot of memory to do its work
so on your big file it uses a HUGE amount of ram which very probably causes your OS to do a lot of swapping which slows down a lot. And there is also some overhead du toto many copyies being made by the `ParLasZipDecompressor`.


So the plan is to fix the implementation of lazrs's `ParLasZipDecompressor`, and if that is not possible we should still be able
to bring back the old wat of decompressing points in `laspy.read` while still keeping the chunk reading feature.

**Comment by tmontaigu** [comment]
laspy 2.0.2 with the latest lazrs version should improve things

**Comment by chrise96** [comment]
Thank you @tmontaigu for improving these libraries! The read time now:
`laspy read in 59.92 seconds.`
